lldb/source/Plugins/Language/CPlusPlus/Generic.cpp

```diff
@@ -16,7 +18,23 @@ lldb::ValueObjectSP lldb_private::formatters::GetDesugaredSmartPointerValue(
 
   auto arg = container_type.GetTypeTemplateArgument(0);
   if (!arg)
-    return ptr.GetSP(); // FIXME: PDB doesn't have info about template arguments
+    // If there isn't enough debug info, use the pointer type as is
+    return ptr.GetSP();
 
   return ptr.Cast(arg.GetPointerType());
 }
+
+lldb_private::SyntheticChildrenFrontEnd *
+lldb_private::formatters::GenericSmartPointerSyntheticFrontEndCreator(
```

Comment:
Looks better, thanks!

Lets move these helpers into `CPlusPlusLanguage` though since they're only used from there (and it feels a tiny bit weird to include the MSVC/Libstc++ headers into `Generic`)

Reply:
Done in https://github.com/llvm/llvm-project/commit/9b88cec72b274360331e7c7bf89e6133b8fbe6d0.